cluster/kubernetes/kubernetes.go

```diff
@@ -402,3 +407,32 @@ func (c *Cluster) ImagesToFetch() registry.ImageCreds {
 
 	return allImageCreds
 }
+
+// getNamespaces returns a list of namespaces that the Flux instance is expected
+// to have access to and can look for resources inside of.
+// It returns a list of all namespaces unless a namespace whitelist has been set on the Cluster
+// instance, in which case it returns a list containing the namespaces from the whitelist
+// that exist in the cluster.
+func (c *Cluster) getNamespaces() ([]apiv1.Namespace, error) {
+	nsList := []apiv1.Namespace{}
+	validNamespaces := map[string]bool{}
+
+	namespaces, err := c.client.Namespaces().List(meta_v1.ListOptions{})
+	if err != nil {
+		return nsList, err
+	}
+
+	for _, namespace := range c.nsWhitelist {
```

Comment:
Very minor optimization / nit: might be better to just do this once on struct creation and store a map instead of a list as a struct member.

Reply:
Good idea, I think that is a sane way to store it.